test/evp_extra_test.c

```diff
@@ -530,6 +532,12 @@ static int test_EVP_SM2(void)
     if (!TEST_ptr(md_ctx_verify = EVP_MD_CTX_new()))
         goto done;
 
+    if (!TEST_true(EVP_DigestSignInit(md_ctx, &sign_ctx, EVP_sm3(), NULL, pkey)))
+        goto done;
+
+    if (!TEST_true(EVP_PKEY_CTX_set_sm2_uid(sign_ctx, "nobody@example.com") > 0))
```

Comment:
Sure, I will add this along with a "real" KAT wired into `evp_test`.

Reply:
Oh well it turned out to be difficult (impossible?) to test a SM2 signature via `evp_test` because it expects all of the hash values to be provided in precomputed form, so instead I added another test to `evp_extra_test` which verifies a known signature created using the personalization field.